app/src/main/java/com/yacgroup/yacguide/database/tourbook/BaseExporter.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Christian Sommer
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.yacgroup.yacguide.database.tourbook
+
+import android.content.ContentResolver
+import android.net.Uri
+import com.yacgroup.yacguide.database.DatabaseWrapper
+import java.io.FileOutputStream
+import java.io.IOException
+
+abstract class BaseExporter {
+    lateinit var contentResolver: ContentResolver
```

Comment:
For me, my approach seemed a bit nicer because it avoids the constructor arguments and feeding them to the base class but it comes with disadvantage of public properties.

Reply:
I agree. Nicer VS safer :) you may also leave it as is. Ok for me

Reply:
OK. I will make it safer.